sql/sql_parse.cc

```diff
@@ -7250,7 +7250,7 @@ bool check_fk_parent_table_access(THD *thd,
                  "REFERENCES",
                  thd->security_ctx->priv_user,
                  thd->security_ctx->host_or_ip,
-                 table_name.str);
+                 db_name.str, table_name.str);
```

Comment:
This is not correct.
1. If `is_qualified_table_name` is false you use `db_name` with uninitialized values.
2. If `is_qualified_table_name` is true, your error message will look something like:
ERROR.... for table `database`.`database.t1`, because table_name string already contains the db_name. See line `7244`.

Please create a test case for both of these situations.